compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1526,6 +1526,12 @@ impl<'a> Parser<'a> {
         if self.token == token::Comma {
             seen_comma = true;
         }
+        if self.eat(&token::Semi) {
+            let sp = self.prev_token.span;
+            let mut err = self.struct_span_err(sp, format!("{adt_ty} fields are separated by `,`"));
+            err.span_suggestion(sp, "replace `;` with `,`", ",", Applicability::MachineApplicable);
```

Comment:
`span_suggestion_short` here will render like "help: replace `;` with `,`" which i think is cleaner